tests/rules/twoslash.test.ts

```diff
@@ -0,0 +1,279 @@
+        column: 7,
+      },
+    ],
+    output: dedent`
+      const four = 4;
+      //    ^? const four: 4
+    `,
+  },
+  // Fixer for comment that doesn't continue the twoslash comment
+  {
+    code: dedent`
+      const square = (x: number) => x * x;
+      const four = square(2);
+      //    ^? const four: string
+    `,
+    optionsSet: [[]],
+    errors: [
+      {
+        messageId: 'TypesDoNotMatch',
+        line: 2,
+        column: 7,
+      },
+    ],
+    output: dedent`
+      const square = (x: number) => x * x;
+      const four = square(2);
+      //    ^? const four: number
+    `,
+  },
+  // Same as above but not the last line of the file
+  {
+    code: dedent`
+      const square = (x: number) => x * x;
+      const four = square(2);
+      //    ^? const four: string
+      // not the last line.
+    `,
+    optionsSet: [[]],
+    errors: [
+      {
+        messageId: 'TypesDoNotMatch',
+        line: 2,
+        column: 7,
+      },
+    ],
+    output: dedent`
+      const square = (x: number) => x * x;
+      const four = square(2);
+      //    ^? const four: number
+      // not the last line.
+    `,
+  },
+  // While whitespace is ignored, field order does matter.
+  {
+    code: dedent`
+      const t = { a: 17, b: 'on' as const };
+      //    ^? const t: {
+      //         b: "on";
+      //         a: number;
+      //       }
+    `,
+    optionsSet: [[]],
+    errors: [
+      {
+        messageId: 'TypesDoNotMatch',
+        line: 1,
+        column: 7,
+      },
+    ],
+    output: dedent`
+      const t = { a: 17, b: 'on' as const };
+      //    ^? const t: {
+      //           a: number;
+      //           b: "on";
+      //       }
+    `,
+  },
+  // A single line assertion can become a multiline after fixing.
+  {
+    code: dedent`
+      const t = { a: 17, b: 'on' as const };
+      //    ^? const t: boolean
+    `,
+    optionsSet: [[]],
+    errors: [
+      {
+        messageId: 'TypesDoNotMatch',
+        line: 1,
+        column: 7,
+      },
+    ],
+    output: dedent`
+      const t = { a: 17, b: 'on' as const };
+      //    ^? const t: {
+      //           a: number;
+      //           b: "on";
+      //       }
+    `,
+  },
+  // A multiline assertion can become a single line after fixing.
+  {
+    code: dedent`
+      let four = 4;
+      //    ^? let four: {
+      //           a: number;
+      //           b: "on";
+      //       }
+    `,
+    optionsSet: [[]],
+    errors: [
+      {
+        messageId: 'TypesDoNotMatch',
+        line: 1,
+        column: 5,
+      },
+    ],
+    output: dedent`
+      let four = 4;
+      //    ^? let four: number
+    `,
+  },
+  {
+    code: dedent`
+      const square = (x: number) => x * x;
+      const four = square(2);
+      // ^? const four: string
+    `,
+    optionsSet: [[]],
+    errors: [
+      {
+        messageId: 'OrphanAssertion',
+        line: 2,
+        column: 1, // would column 3 be better?
```

Comment:
Yeah, I do find it odd to put the squiggly on the node rather than the comment. But that's all existing code - I think it's fine to consider that for a followup change.

```suggestion
        column: 1,
```